test/js/web/fetch/fetch.stream.test.ts

```diff
@@ -0,0 +1,1129 @@
+            },
+          });
+        },
+      });
+
+      let res = await fetch(`http://${server.hostname}:${server.port}`, {});
+      gcTick(false);
+      const reader = res.body?.getReader();
+
+      let buffer = Buffer.alloc(0);
+      let parts = 0;
+      while (true) {
+        gcTick(false);
+
+        const { done, value } = (await reader?.read()) as ReadableStreamDefaultReadResult<any>;
+        if (value) {
+          buffer = Buffer.concat([buffer, value]);
+          parts++;
+        }
+        if (done) {
+          break;
+        }
+      }
+
+      gcTick(false);
+      expect(buffer.toString("utf8")).toBe(content);
+      expect(parts).toBeGreaterThanOrEqual(1);
+    } finally {
+      server?.stop();
+      server_original?.stop();
+    }
+  });
+  const matrix = [
+    { name: "small", data: fixtures["fixture"] },
+    { name: "small text", data: smallText },
+    { name: "big text", data: bigText },
+    { name: "img", data: fixtures["fixture.png"] },
+    { name: "empty", data: empty },
+  ];
+  for (let i = 0; i < matrix.length; i++) {
+    const fixture = matrix[i];
+    for (let j = 0; j < matrix.length; j++) {
+      const fixtureb = matrix[j];
+      const test = fixture.name == "empty" && fixtureb.name == "empty" ? it.todo : it;
+      test(`can handle fixture ${fixture.name} x ${fixtureb.name}`, async () => {
+        let server: Server | null = null;
+        try {
+          //@ts-ignore
+          const data = fixture.data;
+          //@ts-ignore
+          const data_b = fixtureb.data;
+          const content = Buffer.concat([data, data_b]);
+          server = Bun.serve({
+            port: 0,
+            fetch(req) {
+              return new Response(
+                new ReadableStream({
+                  type: "direct",
+                  async pull(controller) {
+                    controller.write(data);
+                    await controller.flush();
+                    await Bun.sleep(100);
+                    controller.write(data_b);
+                    await controller.flush();
+                    controller.close();
+                  },
+                }),
+                {
+                  status: 200,
+                  headers: {
+                    "Content-Type": "text/plain",
+                  },
+                },
+              );
+            },
+          });
+
+          const server_url = `http://${server.hostname}:${server.port}`;
+          const res = await fetch(server_url);
+          const reader = res.body?.getReader();
+          let buffer = Buffer.alloc(0);
+          while (true) {
+            const { done, value } = (await reader?.read()) as ReadableStreamDefaultReadResult<any>;
+            if (value) {
+              buffer = Buffer.concat([buffer, value]);
+            }
+            if (done) {
+              break;
+            }
+          }
+          gcTick(false);
+          expect(buffer.byteLength).toBe(content.byteLength);
```

Comment:
This only checks the length. It needs to check the value too. `expect(buffer).toEqual(content)`

Reply:
Will add it

Reply:
fixed on https://github.com/oven-sh/bun/pull/4127/commits/70d8e6941743d8ae072946846f015751201d1601